xonsh/pyghooks.py

```diff
@@ -281,7 +282,8 @@ def __init__(self, style_name='default'):
         style_name : str, optional
             The style name to initialize with.
         """
-        self.trap = {}  # for custom colors
+        self.trap = {} # for traping custom colors set by user
+        self._mod = {} # for internal modification of the style
```

Comment:
No. I need to clear whatever modification I made when the style changes. Since those modifications depend on the style. But I could maybe just modify the pygments part of the chainmap directly if the class kept a reference to it. Then I could remove the double trap thing...I will try and ping you again...